schemas/2015-08-01/Microsoft.ServiceBus.json

```diff
@@ -0,0 +1,1432 @@
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ]
+        },
+        "resources": {
+          "type": "array",
+          "items": {
+            "oneOf": [
+              {
+                "$ref": "#/definitions/namespaces_topics_subscriptions_childResource"
+              },
+              {
+                "$ref": "#/definitions/namespaces_topics_authorizationRules_childResource"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "type",
+        "apiVersion",
+        "location",
+        "properties"
+      ],
+      "description": "Microsoft.ServiceBus/namespaces/topics"
+    },
+    "namespaces_topics_authorizationRules": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.ServiceBus/namespaces/topics/authorizationRules"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2015-08-01"
+          ]
+        },
+        "location": {
+          "type": "string",
+          "description": "data center location."
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the AuthorizationRule."
+        },
+        "properties": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/SharedAccessAuthorizationRuleProperties"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ]
+        }
+      },
+      "required": [
+        "type",
+        "apiVersion",
+        "properties"
+      ],
+      "description": "Microsoft.ServiceBus/namespaces/topics/authorizationRules"
+    },
+    "namespaces_topics_subscriptions": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.ServiceBus/namespaces/topics/subscriptions"
+          ]
+        },
+        "apiVersion": {
+          "type": "string",
+          "enum": [
+            "2015-08-01"
+          ]
+        },
+        "location": {
+          "type": "string",
+          "description": "Subscription data center location."
+        },
+        "properties": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/SubscriptionProperties"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ]
+        }
+      },
+      "required": [
+        "type",
+        "apiVersion",
+        "location",
+        "properties"
+      ],
+      "description": "Microsoft.ServiceBus/namespaces/topics/subscriptions"
+    }
+  },
+  "definitions": {
+    "MessageCountDetails": {
```

Comment:
one common suggestion is to consider use "required" for the properties if there is any required, it will be helpful for VS intellisense and template export.